### PR TITLE
Esha/profiles context

### PR DESCRIPTION
**Pull request by eshabansiya**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

ProfileProvider context added and onboarding implemented with it.

Q: Able to edit fields even after it is first initialized in onboarding? Not sure if it should be that fields can only be inputed once  when it is first null on this page

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior 
Screenshot 2025-10-12 at 4 31 05 PM

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- you can use esha.bansiya@gmail.com with esha123 to sign in
- and then make changes to onboarding and see if it updates in the supabase

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: app/onboarding/page.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  useEffect(() => {
+    if (profile) {
+      setFirstName(profile.first_name ?? "");
+      setLastName(profile.last_name ?? "");
+      setCountry(profile.country ?? "");
+      setRole(profile.org_role ?? "");
+    }
+  }, [profile]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```
Comment (me-liu) [inline app/onboarding/page.tsx]: please move this function to the provider page! pref called handleProfileSubmit

File: utils/ProfileProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          .select("*")
+          .eq("id", uid)
+          .single();
+
+        setProfile(
+          row ?? {
+            id: uid,
+            first_name: null,
+            last_name: null,
+            country: null,
```
Comment (me-liu) [inline utils/ProfileProvider.tsx]: make sure the other fields are being set as well! so the user_type, user_group id, etc.

File: app/onboarding/page.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    event.preventDefault();
+    setFormMessage("");
+    setSave(true);
+
+    await supabase.auth.getSession();
```
Comment (me-liu) [inline app/onboarding/page.tsx]: this line may not be necessary bc you fetch from your context earlier

**Review by me-liu — COMMENTED** [review]
very sexy things look good except for minor changes

File: app/auth/login/page.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
+"use client";
```
[the remark this answers is not in the document]
Comment (eshabansiya) [reply app/auth/login/page.tsx]: yes, but also so that the onboarding would work, but also it kept my auth session i think so it was fine but yea this was after i pulled from main i think, this was from dylans sprint

File: app/onboarding/page.tsx (hunk 1 above)
Comment (eshabansiya) [reply app/onboarding/page.tsx]: is this move the handleSubmit or the getting the Profile to see if its null?
